sesame2spiner/generate_files.cpp

```diff
@@ -171,12 +174,27 @@ herr_t saveAllMaterials(const std::string &savename,
 
   for (size_t i = 0; i < matids.size(); i++) {
     int matid = matids[i];
+    if (used_matids.count(matid) > 0) {
+      std::cerr << "...Duplicate matid " << matid << " detected. Skipping."
+		<< std::endl;
+      continue;
+    }
+    used_matids.insert(matid);
 
     std::cout << "..." << matid << std::endl;
 
     eosGetMetadata(matid, metadata, Verbosity::Debug);
     if (printMetadata) std::cout << metadata << std::endl;
     std::string name = params[i].Get("name", metadata.name);
+    if (used_names.count(name) > 0) {
+      std::string new_name = "material_" + std::to_string(i);
+      std::cerr << "...WARNING: Name " << name << " already used. "
+		<< "using a default name: "
+		<< new_name
+		<< std::endl;
+      name = new_name;
```

Comment:
what about using the repeated name + an integer @Yurlungur ?

Reply:
@Yurlungur I'll approve once we resolve how to move forward on this.

Reply:
Yeah, name + integer is fine. I did it this way because it's easier than tracking an integer count, but integer count should be easy too. Been under the weather a few days. Will do this when I'm at my desk again.

Reply:
maybe we use a default name if the "reused name" is an empty string, but if there is an actual name that is re-used, preserving some of that information could be useful to downstream users.

Reply:
I'm not sure I can catch all the possible "empty string" names but I'll try.

Reply:
@dholladay00 the new version does the following:
- If the name is an empty string or the string "-1", which I noticed some unnamed materials have in their name field, rename the material "material_n" where n is the number of the input deck passed into sesame2spiner
- If matids are repeated, skips the duplicate matid
- If two materials have the same, non-trivial name, `sesame2spiner` will enumerate them. i.e., if you had three materials named "metal," then in the sp5 file they'll be called `metal`, `metal_2`, `metal_3`.

Give it a try.

Reply:
This works great now!